scipy/integrate/_ivp/tests/test_ivp.py

```diff
@@ -683,6 +687,41 @@ def test_t_eval_dense_output():
     assert_(np.all(e < 5))
 
 
+# This test ensures that the interpolant calculated through LSODA is correct.
+# Arguments provided to solve_ivp should allow for method order changes within
+# the solver, thereby guaranteeing a full test coverage of _dense_output_impl
+# in lsoda.py
+def test_dense_output_LSODA():
+    rtol = 1e-3
+    atol = 1e-6
+    y0 = [0.05]
+    t_span = [-2, 2]
+    first_step = 1e-3
+    res = solve_ivp(fun_dense_output_LSODA, t_span, y0, method="LSODA",
+                    dense_output=True, first_step=first_step, max_step=1,
+                    rtol=rtol, atol=atol, jac=jac_dense_output_LSODA)
+
+    assert_equal(res.t[0], t_span[0])
+    assert_equal(res.t[-1], t_span[-1])
+    assert_allclose(first_step, np.abs(res.t[1] - t_span[0]))
+    assert_(res.t_events is None)
```

Comment:
for *new* testing code, we have a modest preference for plain `assert`

Reply:
Is it fine to replace all instances in the file with the plain `assert`? I have done it for consistency.

Reply:
In the future, please avoid mixing wholesale changes like this with another PR. The reason is just practical - GH shows all of the changes, and they distract from the point of the PR.